EIPS/eip-7405.md

```diff
@@ -0,0 +1,218 @@
+---
+eip: 7405
+title: Portable Smart Contract Accounts
+description: A solution working at the Proxy (ERC-1967) layer, allowing users to easily migrate their SCAs between different wallets.
+author: Aaron Yee (@aaronyee-eth)
+discussions-to: https://ethereum-magicians.org/t/erc-7405-portable-smart-contract-accounts/15236
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-07-26
+requires: 1967
+---
+
+## Abstract
+
+The proposed EIP introduces a standardized solution called "Portable Smart Contract Accounts" (PSCA) that aims to address the lack of portability and compatibility issues faced by Smart Contract Accounts (SCA) across different wallet providers. Based on **[ERC-1967](./eip-1967.md)**, the PSCA system allows users to easily migrate their SCAs between different wallets using new, randomly generated migration keys. This provides a similar experience to exporting an externally owned account (EOA) with a private key or mnemonic. The system ensures security by employing signatures and time locks, allowing users to verify and cancel migration operations during the lock period, thereby preventing potential malicious actions. PSCA offers a non-intrusive and cost-effective approach, enhancing the interoperability and composability within the Account Abstraction (AA) ecosystem.
+
+## Motivation
+
+With the introduction of the **[ERC-4337](./eip-4337.md)** standard, Account Abstraction (AA) related infrastructure and Smart Contract Accounts (SCA) have been widely adopted in the community. However, unlike externally owned accounts (EOA), SCAs have a more diverse code space, leading to varying contract implementations across different wallet providers. Consequently, the lack of portability for SCAs has become a significant issue, making it challenging for users to migrate their accounts between different wallet providers. While **[ERC-6900](./eip-6900.md)** proposed a modular approach for SCA accounts, it comes with higher implementation costs and specific prerequisites for wallet implementations.
```

Comment:
Small warning: if you link to another proposal, it limits how far your proposal can advance. For example, ERC-6900 is currently a Draft, which means this proposal cannot advance to Review until ERC-6900 does.

If you're okay waiting for ERC-6900, you can leave the reference in. Otherwise, just remove it.

Reply:
Okay